static/src/javascripts/projects/membership/stripe.js

```diff
@@ -0,0 +1,139 @@
+     *  card:
+     *   -last4: string of last 4 digits of cc
+     *   -type:  string of card type (capitalised)
+     *   */
+
+    function display(parent, card) {
+        var $parent = $(parent);
+        var $number = $('.js-manage-account-card', $parent);
+        var $last4 = $('.js-manage-account-card-last4', $parent);
+        var $type = $('.js-manage-account-card-type', $parent);
+        var $button = $('.js-manage-account-change-card', $parent);
+        var $updating = $('.js-updating', $parent);
+
+        /*  show/hide */
+        var loading = function () {
+            var HIDDEN = 'is-hidden';
+            var $elems = [$button, $number, $type, $last4];
+            var stack = 1; //We're waiting for the card to be displayed.
+            return function (isLoading) {
+                fastdom.write(function () {
+                    if (isLoading) {
+                        stack++;
+                    } else {
+                        stack--;
+                    }
+                    if (stack > 0) {
+                        $elems.forEach(function ($e) {
+                            $e.addClass(HIDDEN);
+                        });
+                        $updating.removeClass(HIDDEN);
+
+                    } else {
+                        $elems.forEach(function ($e) {
+                            $e.removeClass(HIDDEN);
+                        });
+                        $updating.addClass(HIDDEN);
+                    }
+                });
+            };
+        }();
+
+        var oldCardType = $type.data('type');
+        var newCardType = 'i-' + card.type.toLowerCase().replace(' ', '-');
+
+        if (oldCardType) {
+            $type.removeClass(oldCardType);
+        }
+
+        fastdom.write(function () {
+            $last4.text(card.last4);
+            loading(false);
+            $type.addClass(newCardType);
+            $type.data('type', newCardType);
+            $parent.removeClass('is-hidden');
+        });
+
+        bean.off($button[0], 'click');
+        bean.on($button[0], 'click', handler());
+
+
+        /*
+         * Closes over the event handler for the Change Card button
+         */
+        function handler() {
+
+            var product = $parent.data('product');
+            var endpoint = config.page.userAttributesApiUrl + '/me/' + product + '-update-card';
+            var email = $button.data('email');
+            return function (e) {
+                e.preventDefault();
+                loading(true);
+
+                checkoutHandler.open({
+                    email: email,
+                    description: 'Update your card details',
+                    panelLabel: 'Update',
+                    token: update(endpoint),
+                    closed: function () {
+                        loading(false);
+                    }
+
+                });
+                // Close Checkout on page navigation:
+                window.addEventListener('popstate', function () {
```

Comment:
It should be, but: https://stripe.com/docs/checkout#integration-custom
![image](https://cloud.githubusercontent.com/assets/2670496/20058939/96d24c68-a4eb-11e6-885e-7db506c14fac.png)